modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -149,20 +149,41 @@ func (p *Processor) consume(resourceSpans []*v1_trace.ResourceSpans) (err error)
 
 		for _, ils := range rs.InstrumentationLibrarySpans {
 			for _, span := range ils.Spans {
+				connectionType := store.Unknown
+
 				switch span.Kind {
+				case v1_trace.Span_SPAN_KIND_PRODUCER:
+					connectionType = store.MessagingSystem
+					fallthrough
```

Comment:
I think we should add a comment here to emphasise why the producer case was placed before client and similarly for consumer <> server.

Reply:
Whoa, I've never seen `fallthrough` before.  Neat.